pipeline/trusted-resources/cmd/sign/main.go

```diff
@@ -119,3 +118,25 @@ func Sign(ctx context.Context, tr *v1beta1.TaskRun, signer sigstore.Signer, writ
 	_, err = writer.Write(signedBuf)
 	return err
 }
+
+// Sign the taskrun and output signed bytes to writer
+func SignTask(ctx context.Context, task *v1beta1.Task, signer sigstore.Signer, writer io.Writer) error {
+	sig, err := trustedtask.SignInterface(signer, task)
+	if err != nil {
+		return err
+	}
+
+	if task.Annotations == nil {
+		task.Annotations = map[string]string{trustedtask.SignatureAnnotation: sig}
+	} else {
+		task.Annotations[trustedtask.SignatureAnnotation] = sig
+	}
+
+	signedBuf, err := yaml.Marshal(task)
+	if err != nil {
+		return err
+	}
+
+	_, err = writer.Write(signedBuf)
+	return err
+}
```

Comment:
You should be able to combine these by using the [object interface](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#Object)